spec/javascripts/Stats/lib/csv_link.spec.js

```diff
@@ -15,18 +15,28 @@ describe('StatsCSVLink', () => {
         '2001-05-11T07:00:00+00:00'
       ],
       [
-        'zaphod',
+        'with "quotes"',
         '12',
         '30'
       ],
       [
-        'marvin',
+        'marvin #2',
         '11',
         '31'
+      ],
+      [
+        'with, comma',
+        '10',
+        '32'
       ]
     ]
   }
-  const expectedCsvString = 'datetime,zaphod,marvin%0A11 Mar 1952 07:00:00 GMT,12,11%0A11 May 2001 08:00:00 BST,30,31'
+  const cells = [
+    ['datetime', '"with ""quotes"""', 'marvin #2', '"with, comma"'],
+    ['11 Mar 1952 07:00:00 GMT', '12', '11', '10'],
+    ['11 May 2001 08:00:00 BST', '30', '31', '32']
+  ]
+  const expectedCsvString = cells.map(row => row.map(cell => encodeURIComponent(cell)).join(',')).join('%0A')
```

Comment:
This "expected string" assignment might be a bit ridiculous, as it's basically replicating what the app logic itself does... But the thing is that if we provide just the expected string, it is completely unreadable:
```
datetime,%22with%20%22%22quotes%22%22%22,marvin%20%232,%22with%2C%20comma%22%0A11%20Mar%201952%2007%3A00%3A00%20GMT,12,11,10%0A11%20May%202001%2008%3A00%3A00%20BST,30,31,32
```